### PR TITLE
Switch teleport example to real world scenario

### PR DESCRIPTION
**Pull request by marina-mosti**
Issue #321

### CODE REVIEW

File: src/guide/teleport.md [hunk 1]
```diff
@@ -2,53 +2,99 @@
 
 Vue encourages us to build our UIs by encapsulating UI and related behavior into components. We can nest them inside one another to build a tree that makes up an application UI.
 
-However, sometimes a part of a component's template belongs into this component logically, while from a technical point of view, it would be preferable to move this part of the template somewhere else in the DOM, outside of Vue app. For example, due to styling requirements, we want to move `<p id="content">` from it's deeply nested position to the `<div>` with `id="endofbody"`
+However, sometimes a part of a component's template belongs into this component logically, while from a technical point of view, it would be preferable to move this part of the template somewhere else in the DOM, outside of the Vue app. 
+
+A common scenario for this, is creating a component that includes a full-screen modal. In most cases, you'd want the modal's logic to live within the component, but the positioning of the modal quickly becomes a CSS nightmare due to the positioning of the parent elements.
```
Comment (NataliaTepluhina) [inline src/guide/teleport.md:7]: **suggestion**: let's change `CSS nightmare` to something less radical

```suggestion
A common scenario for this is creating a component that includes a full-screen modal. In most cases, you'd want the modal's logic to live within the component, but the positioning of the modal quickly becomes a challenging CSS task due to the positioning of the parent elements. 
```

**Comment by bencodezen** [comment]
Great job @marina-mosti! Once the comments from @NataliaTepluhina are addressed it is a 👍 from me!

**Comment by marina-mosti** [comment]
Thanks for the review @NataliaTepluhina and @bencodezen 
I've applied the suggestions and modified the codepen, it has to be copied over to the Vue account if approved though.

https://codepen.io/immarina/pen/QWyPmOj

File: src/guide/teleport.md [hunk 2]
```diff
@@ -2,53 +2,95 @@
...
+```
+
+As a result, once we click the button to open the modal, Vue will correctly render the modal's content as a child of the `body` tag.
+
+<p class="codepen" data-height="300" data-theme-id="39028" data-default-tab="js,result" data-user="Vue" data-slug-hash="PoZgggm" data-preview="true" data-editable="true" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Teleport">
+  <span>See the Pen <a href="https://codepen.io/team/Vue/pen/PoZgggm">
+  Vue 3 Teleport</a> by Vue (<a href="https://codepen.io/Vue">@Vue</a>)
   on <a href="https://codepen.io">CodePen</a>.</span>
 </p>
 <script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```
Comment (NataliaTepluhina) [inline src/guide/teleport.md:92]: Added a forked codepen here

```suggestion
<p class="codepen" data-height="300" data-theme-id="39028" data-default-tab="js,result" data-user="Vue" data-slug-hash="gOPNvjR" data-preview="true" data-editable="true" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Teleport">
  <span>See the Pen <a href="https://codepen.io/team/Vue/pen/gOPNvjR">
  Vue 3 Teleport</a> by Vue (<a href="https://codepen.io/Vue">@Vue</a>)
  on <a href="https://codepen.io">CodePen</a>.</span>
</p>
<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```

File: src/guide/teleport.md [hunk 3]
```diff
@@ -2,53 +2,95 @@
...
+      modalOpen: false
+    }
+  }
+})
 ```
 
-As a result, we will have `teleport` content moved in the rendered DOM tree:
+When using this component inside the initial HTML structure, we can see a problem - the modal is being rendered inside the deeply nested `div` and the `position: absolute` of the modal takes the parent relatively positioned `div` as reference.
+
+Teleport provides a clean way to allow us to control under which parent in our DOM we want a piece of HTML to be rendered at, without having to resort to global state or splitting this into two components.
```
Comment (NataliaTepluhina) [inline src/guide/teleport.md:54]: **nitpick**: small grammar fix

```suggestion
Teleport provides a clean way to allow us to control under which parent in our DOM we want a piece of HTML to be rendered, without having to resort to global state or splitting this into two components.
```

**Comment by marina-mosti** [comment]
Thanks for the double check @NataliaTepluhina 
I think were good to go

**Pull request merged**